R/module-helpers.R

```diff
@@ -85,3 +85,171 @@ rename_SO2 <- function(x, so2_map, is_awb = FALSE) {
+#' @param names Column names to use out of data, character
+#' @param base_header Base header that is used for the logit type tables which will get
+#'  pasted with what the logit.type for each table, character
+#' @param default_logit_type Default logit function to use if the user did not specify one, character
+#' @param include_equiv_table Include EQUIV_TABLE as well? Logical
+#' @param write_all_regions Write each table to all regions? Logical
+#' @param ... Other parameters to pass to \code{\link{write_to_all_regions}}
+#' @details The data has to be partitioned into multiple tables, one for each logit.type. Each element of
+#' the returned list has two variables: 1) the header for the table, and 2) the data for the table.
+#' If requested, an additional \code{EQUIV_TABLE} table will be added so that the tables that contain the logit
+#' exponents do not themselves have to know what logit.type they are using and thus do not need to be
+#' partitioned.
+#' @return The tables TODO NEEED BETTER DOCUMENTATION
+get_logit_fn_tables <- function(data, names, base_header,
+                                include_equiv_table = TRUE,
+                                write_all_regions = FALSE,
+                                default_logit_type = gcam.LOGIT_TYPES[1], ...) {
+  assert_that(is_tibble(data))
+  assert_that(is.character(names))
+  assert_that(is.character(base_header))
+  assert_that(is.logical(include_equiv_table))
+  assert_that(is.logical(write_all_regions))
+  assert_that(is.character(default_logit_type))
+
+  # Set the logit type to the default if currently unspecified
+  data$logit.type[is.na(data$logit.type)] <- default_logit_type
+
+  # Note it is safer to create tables for all valid logit types rather than just the
+  # ones included in unique(data$logit.type) even if it results in an empty table
+  # since if we switch all from one type to the other and do not clean out the level
+  # 2 CSV and batch file we will be left with both defined for all rows which is bad.
+
+  tables <- list()
+
+  # Create the EQUIV_TABLE table which allows the Model Interface to be ambiguous about what the
+  # actual logit type is when setting the logit exponent.
+  if(include_equiv_table) {
+    tables[[gcam.EQUIV_TABLE]]$header <- gcam.EQUIV_TABLE
+    d <- tibble(group.name = "LogitType")
+    tag <- 1
+    for(lname in c("dummy-logit-tag", gcam.LOGIT_TYPES)) {
+      d[paste0("tag", tag)] <- lname
+      tag <- tag + 1
+    }
+    tables[[gcam.EQUIV_TABLE]]$data <- d
+  }
+
+  # Loop through each of the logit types and create a table for it
+  # using the appropriate header name and writing to all regions if requested.
+  for(curr_logit_type in gcam.LOGIT_TYPES) {
+    tables[[curr_logit_type]]$header <- paste0(base_header, curr_logit_type)
+    currdata <- data[data$logit.type == curr_logit_type,]
+    if(write_all_regions) {
+      if(nrow(currdata) > 0) {
+        currdata <- write_to_all_regions(currdata, names, ...)
```

Comment:
`write_to_all_regions` needs to specify `GCAM_region_names`. When use separately, `write_to_all_regions(data, names, GCAM_region_names = GCAM_region_names)`, first is the parameter and second is the tibble. But here the tibble is not specified, generates error: `argument "GCAM_region_names" is missing, with no default`.